design/README.md

```diff
@@ -0,0 +1,103 @@
+|:-------------|:-------------|
+<div style="height:8px"></div>
+
+| **Language**     | **Package**        |
+|:-------------|-----------------|
+|![Python Logo](https://docs.aws.amazon.com/cdk/api/latest/img/python32.png) Python|`aws_solutions_constructs.aws_route53_cloudfront`|
+|![Typescript Logo](https://docs.aws.amazon.com/cdk/api/latest/img/typescript32.png) Typescript|`@aws-solutions-constructs/aws-route53-cloudfront`|
+|![Java Logo](https://docs.aws.amazon.com/cdk/api/latest/img/java32.png) Java|`software.amazon.awsconstructs.services.route53cloudfront`|
+
+This AWS Solutions Construct implements an AWS CloudFront distribution against a custom domain and protected with SSL certificate. It lets you specify overrides for CloudFront distribution props so that you can use this pattern with different types of origins.
+
+Here is a minimal deployable pattern definition in Typescript. It assumes that you have a domain purchased from Route53, and hence a corresponding hosted zone already existing.
+
+``` typescript
+import route53 from '@aws-cdk/aws-route53';
+import { Route53ToCloudFront } from '@aws-solutions-constructs/aws-route53-cloudfront';
+
+const hostedZone = route53.HostedZone.fromHostedZoneAttributes(this, 'HostedZone', {
+        hostedZoneId: 'Z00382491YVBQ98RPC5W9',
+        zoneName: 'example.com',
+    });
+
+new Route53ToCloudFront(this, 'test-route53-cloudfront', {
+        existingHostedZoneObj: hostedZone
+    });
+
+```
+
+## Initializer
+
+``` text
+new Route53ToCloudFront(scope: Construct, id: string, props: Route53ToCloudFrontProps);
+```
+
+_Parameters_
+
+* scope [`Construct`](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_core.Construct.html)
+* id `string`
+* props [`Route53ToCloudFrontProps`](#pattern-construct-props)
+
+## Pattern Construct Props
+
+| **Name**     | **Type**        | **Description** |
+|:-------------|:----------------|-----------------|
+|existingHostedZoneObj?|[`route53.HostedZone`](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-route53.HostedZone.html)|Existing instance of Route53 hosted zone object. If it is not provided, a hosted zone will be created based on the given domainName.|
+|domainName?|`string`|The domain name to be used while creating the hosted zone. This is used only if existingHostedZoneObj is not provided.|
+|hostedZoneProps?|[`route53.HostedZoneProps`](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-route53.HostedZoneProps.html)|Optional user provided props to override the default props for the Route53 hosted zone.|
+|certificateArn?|`string`|ARN of an existing certificate from us-east-1 region in ACM. If it is not provided, a certificate will be automatically provisioned for the domain using ACM's DNS based validation. <span style="color:orange">***Please note that the hosted zone needs to be reachabe from the internet in order for the automatic validation to be successful. Please see the [footnotes](#footnotes) for more details.***|</span>
```

Comment:
This is our first construct featuring Route53, so let's think pretty hard about how the Route53 section should look in the Design Guide. Specifically, what construction props and Construct properties are appropriate to set as the standard for Route53 in any construct moving forward.